File: UtilWeDie-Neo-1.21.1/ModsSourceCode/sculkhorde/neoforge-1.21.1/src/main/java/com/github/sculkhorde/datagen/DataGenerators.java
package com.github.sculkhorde.datagen;

import com.github.sculkhorde.core.ModBlocks;
import com.github.sculkhorde.core.SculkHorde;
import net.minecraft.core.HolderLookup;
import net.minecraft.data.DataGenerator;
import net.minecraft.data.DataProvider;
import net.minecraft.data.PackOutput;
import net.minecraft.data.loot.LootTableProvider;
import net.minecraft.resources.ResourceKey;
import net.minecraft.world.level.block.Block;
import net.minecraft.world.level.storage.loot.LootTable;
import net.minecraft.world.level.storage.loot.parameters.LootContextParamSets;
import net.neoforged.neoforge.common.data.ExistingFileHelper;
import net.neoforged.neoforge.data.event.GatherDataEvent;
import net.neoforged.bus.api.SubscribeEvent;
import net.neoforged.fml.common.EventBusSubscriber;

import java.util.List;
import java.util.concurrent.CompletableFuture;
import java.util.stream.Collectors;

@EventBusSubscriber(modid = SculkHorde.MOD_ID, bus = EventBusSubscriber.Bus.MOD)
public class DataGenerators {

    @SubscribeEvent
    public static void gatherData(GatherDataEvent event) {
        DataGenerator generator = event.getGenerator();
        PackOutput packOutput = generator.getPackOutput();
        ExistingFileHelper existingFileHelper = event.getExistingFileHelper();
        CompletableFuture<HolderLookup.Provider> lookupProvider = event.getLookupProvider();

        //client
        generator.addProvider(event.includeClient(), (DataProvider.Factory<ModBlockModelsProvider>) output -> new ModBlockModelsProvider(output, SculkHorde.MOD_ID, existingFileHelper));
        generator.addProvider(event.includeClient(), (DataProvider.Factory<ModBlockStateProvider>) output -> new ModBlockStateProvider(output, SculkHorde.MOD_ID, existingFileHelper));

        //server
        generator.addProvider(event.includeServer(), (DataProvider.Factory<ModGlobalLootModifiersProvider>) output -> new ModGlobalLootModifiersProvider(output, lookupProvider));
        generator.addProvider(event.includeServer(), (DataProvider.Factory<LootTableProvider>) output -> new LootTableProvider(
                output,
                ModBlocks.BLOCKS_TO_DATAGEN.stream().map(pair -> ((Block) pair.getA().get()).getLootTable()).collect(Collectors.toUnmodifiableSet()),
                List.of(new LootTableProvider.SubProviderEntry(
                        ModBlockLootTableSubProvider::new,
                        LootContextParamSets.BLOCK
                )),
                lookupProvider
        ));
    }
}

File: SocialWill/sw_src/src/main/kotlin/com/atsuishio/superbwarfare/datagen/DataGenerators.kt
package com.atsuishio.superbwarfare.datagen

import com.atsuishio.superbwarfare.Mod
import net.neoforged.bus.api.SubscribeEvent
import net.neoforged.fml.common.EventBusSubscriber
import net.neoforged.neoforge.data.event.GatherDataEvent

@EventBusSubscriber(modid = Mod.MODID)
object DataGenerators {
    @SubscribeEvent
    fun gatherData(event: GatherDataEvent) {
        val generator = event.generator
        val packOutput = generator.packOutput
        val lookupProvider = event.lookupProvider
        val existingFileHelper = event.existingFileHelper

        generator.addProvider(event.includeServer(), ModLootTableProvider.create(packOutput, lookupProvider))
        generator.addProvider(event.includeServer(), ModRecipeProvider(packOutput, lookupProvider))
        generator.addProvider(event.includeClient(), ModBlockStateProvider(packOutput, existingFileHelper))
        generator.addProvider(event.includeClient(), ModItemModelProvider(packOutput, existingFileHelper))
        val tagProvider = generator.addProvider(
            event.includeServer(),
            ModBlockTagProvider(packOutput, lookupProvider, existingFileHelper)
        )
        generator.addProvider(
            event.includeServer(),
            ModItemTagProvider(packOutput, lookupProvider, tagProvider.contentsGetter(), existingFileHelper)
        )
        generator.addProvider(
            event.includeServer(),
            ModEntityTypeTagProvider(packOutput, lookupProvider, existingFileHelper)
        )
        generator.addProvider(
            event.includeServer(),
            ModDamageTypeTagProvider(packOutput, lookupProvider, existingFileHelper)
        )
        generator.addProvider(
            event.includeServer(),
            ModAdvancementProvider(packOutput, lookupProvider, existingFileHelper)
        )
        generator.addProvider(event.includeServer(), ModPerkTagProvider(packOutput, lookupProvider, existingFileHelper))
        generator.addProvider(event.includeServer(), ModWreckageLootProvider(packOutput, existingFileHelper))
    }
}
